File: frontend/src/api/plan.ts
import request from '../utils/request';

export interface Plan {
    id: number;
    patient_id: number;
    goal: string;
    phases: Array<{
        phase: string;
        actions: Record<string, string[]>;
    }>;
    generated_at_jieqi: string;
}

export interface Survey {
    id: number;
    patient_id: number;
    survey_type: string;
    answers: Record<string, number | string>;
    score: number;
    summary_advice: string;
    created_at: string;
}

export interface SurveyCreate {
    patient_id: number;
    survey_type: string;
    answers: Record<string, number | string>;
}

export interface TimelineEvent {
    id: number;
    patient_id: number;
    event_type: string;
    title: string;
    description?: string;
    occurred_at: string;
}

export const generatePlan = async (patientId: number): Promise<Plan> => {
    return request.post('/plans', null, { params: { patient_id: patientId } });
};

export const getLatestPlan = async (patientId: number): Promise<Plan> => {
    return request.get('/plans/latest', { params: { patient_id: patientId } });
};

export const getTimeline = async (patientId: number): Promise<TimelineEvent[]> => {
    return request.get('/timeline/', { params: { patient_id: patientId } });
};

// Survey APIs
export const getSurveys = async (patientId: number): Promise<Survey[]> => {
    return request.get('/surveys', { params: { patient_id: patientId } });
};

export const submitSurvey = async (data: SurveyCreate): Promise<Survey> => {
    return request.post('/surveys', data);
};

File: backend/app/models/plan.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, JSON, Text
from sqlalchemy.orm import relationship
from datetime import datetime
from app.models.base import Base

class Plan(Base):
    __tablename__ = "plans"
    
    id = Column(Integer, primary_key=True, index=True)
    patient_id = Column(Integer, ForeignKey("patient.id"))
    
    # Context
    generated_at_jieqi = Column(String) # e.g., "Dongzhi"
    start_date = Column(DateTime)
    end_date = Column(DateTime)
    
    goal = Column(String) # e.g. "Stabilize and Hydrate"
    
    # Structure: [ { "phase": "Week 1-2", "actions": {...} }, ... ]
    phases = Column(JSON)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    
    patient = relationship("Patient", back_populates="plans")

class Survey(Base):
    __tablename__ = "surveys"
    
    id = Column(Integer, primary_key=True, index=True)
    patient_id = Column(Integer, ForeignKey("patient.id"))
    
    survey_type = Column(String) # "Health" or "Satisfaction"
    
    # JSON content of answers
    answers = Column(JSON)
    
    # Calculated Score
    score = Column(Integer)
    summary_advice = Column(Text)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    
    patient = relationship("Patient", back_populates="surveys")

class TimelineEvent(Base):
    __tablename__ = "timeline_events"

    id = Column(Integer, primary_key=True, index=True)
    patient_id = Column(Integer, ForeignKey("patient.id"))
    
    event_type = Column(String) # "Exam", "Plan", "Survey", "Visit", "Purchase"
    title = Column(String)
    description = Column(String, nullable=True)
    related_id = Column(Integer, nullable=True) # ID of the related Exam/Plan/Survey
    
    occurred_at = Column(DateTime, default=datetime.utcnow)
    
    patient = relationship("Patient", back_populates="timeline_events")
